Monika-After-Story/MonikaModDev
# Additions to the island event

### Additions:
- `MASAudioData` - isn't used as of now, but will be used later for the next wave of changes

### Changes:
- `play_song` is deprecated in favour of `mas_play_song`
- various internal functions (island-related) renamed to snake_case to be consistent with the rest of the funcs in the namespace
- backgrounds now properly report which images are missing for which filters
- cleanup of the background framework
- - new internal funcs
- - - `getBackground`
- - - `unlockBackground`
- - - `lockBackground`
- - - `isBackgroundUnlocked`
- - new global funcs
- - - `mas_getCurrentBackgroundId`
- - - `mas_isBackgroundUnlocked`
- background selector was improved
- - we don't suggest to select the current background anymore
- - replaced duplicated code with predicate functions for filtering
- - in case no backgrounds are available, we **will** show the default as a fallback option (you still can't go into the same background)

### Fixes:
- fixed a typo that lead to a name error in `_mfwm_find_fb_def`

### Recognition:
- new deco by @Nemu-sus
- new dlg by @lunulae 

### Testing:
- verify playing music still works
- verify the background selector still works
- - and you can't go into the same room
- - and current room isn't being suggested
- - and on O31 only the rooms with the required deco are shown
- verify the islands still work
- - test during different time of day (filter) and weather
- - test on O31 (don't forget about the update script)
- - - again, test during the day and night
- - verify your unlocks are valid
- - verify the new dialogue (on click events) is good
